### PR TITLE
#4800: Resources: set cache-control-header instead of expires-header (JSF 2.3)

### PR DESCRIPTION
**Pull request by christophs78**
Signed-off-by: Christoph Straßer <christoph.strasser78@gmail.com>

### CODE REVIEW

**Comment by christophs78** [comment]
Hi @arjantijms!
My first contribution to mojarra...
Due to some reasons Travis CI does not kick in. (even after 6 hours) Travis-job is "received" but does not get queued.
image
Maybe we should think about moving from Travis CI to Github Action. We did this with PrimeFaces (and addon-projects) in october which worked quite well. (https://github.com/primefaces/primefaces/issues/6391)

**Comment by arjantijms** [comment]
Hi Chris, thanks for the contribution! Travis is currently not in the best condition. Most EE4J projects use the Eclipse provided Jenkins for CI though.

@BalusC can you review this one?

File: impl/src/main/java/com/sun/faces/application/resource/ResourceImpl.java [hunk 1]
```diff
@@ -188,16 +188,11 @@ public URL getURL() {
                 responseHeaders = new HashMap<>(6, 1.0f);
             }
 
-            long expiresTime;
             if (FacesContext.getCurrentInstance().isProjectStage(Development)) {
-                expiresTime = new Date().getTime();
+                responseHeaders.put("Cache-Control", "no-cache");
```
Comment (BalusC) [inline impl/src/main/java/com/sun/faces/application/resource/ResourceImpl.java:192]: Should be `no-store, must-revalidate`.

Comment (christophs78) [reply impl/src/main/java/com/sun/faces/application/resource/ResourceImpl.java:192]: Ok, I changed this.
IMO this is to some degree a matter of taste. Even englisch and german MDN - page for https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control prefer different solutions.

Comment (BalusC) [reply impl/src/main/java/com/sun/faces/application/resource/ResourceImpl.java]: Not a matter of taste. If the resource was already (accidentally) cached, `no-cache` doesn't wipe it.

**Pull request merged**